### ISSUE TITLE
Changes in CI configurations

### ISSUE THREAD

**Issue by pouladpld** [TelegramBots/telegram.bot#421]
@MrRoundRobin
 I've set up the CI builds on AppVeyor and Travis CI. 

**AppVeyor** builds (any branch updates) and publishes the MyGet package (only if `develop` branch) as before.

**Travis CI**, which cares only about `master` and `develop` branches, builds the project with _Release_ configurations and immediately runs the systems integration tests. I created a super group chat for that and respond to my test bot's messages to get the tests completed.

It's not easy to build SandCastle project on CI servers as it needs specific tools and setup so I moved it to a protected `docs` branch which is excluded from CI builds. We can merge changes from `master` into `docs` in order to generate documentations.

Please change the MyGet configurations in [appveyor.yml](https://github.com/TelegramBots/telegram.bot/blob/develop/appveyor.yml) to publish `Telegram.Bot` package instead of `.Core`.

**Comment by pouladpld** [comment]
In addition, an unit test project is added and tests run on AppVeyor build server only.

**Comment by pouladpld** [comment]
@MrRoundRobin Travis-CI is already configured. It'll be fine after #435 is merged into `master`.
